stream.go

```diff
@@ -54,15 +56,18 @@ func newSendStream(str quic.SendStream, hdr []byte, onClose func()) *sendStream
 	return &sendStream{str: str, streamHdr: hdr, onClose: onClose}
 }
 
-func (s *sendStream) maybeSendStreamHeader() error {
-	if len(s.streamHdr) == 0 {
-		return nil
-	}
-	if _, err := s.str.Write(s.streamHdr); err != nil {
-		return err
-	}
-	s.streamHdr = nil
-	return nil
+func (s *sendStream) maybeSendStreamHeader() (err error) {
+	s.once.Do(func() {
+		if len(s.streamHdr) == 0 {
+			return
+		}
```

Comment:
You don't need this check any more, do you?

Reply:
I don't. Removed it.